### PR TITLE
Add test to sync repo having zst compression

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding a test to sync repository having zst type of compression
Closed loop BZ#2241934

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repositories.py -k test_positive_sync_upstream_repo_with_zst_compression

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6062
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_repositories.py -k test_positive_sync_upstream_repo_with_zst_compression --external-logging
Test Result : =========== 1 passed, 10 deselected, 1 warning in 776.43s (0:12:56) ============
```

File: tests/foreman/api/test_repositories.py [hunk 1]
```diff
@@ -186,6 +191,40 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
...
+        1. Sync upstream repository having zst type compression.
+        2. Assert that no errors related to compression type  are present in
+            /var/log/messages.
+        3. Assert that sync was executed properly.
+
+    :CaseComponent: Pulp
+
+    :BZ: 2241934
+    """
+    product = target_sat.api.Product(organization=module_org).create()
```
Comment (shweta83) [inline tests/foreman/api/test_repositories.py:214]: you can use module_product here also

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repositories.py -k test_positive_sync_upstream_repo_with_zst_compression

**Review by Griffin-Sullivan — APPROVED** [review]

File: tests/foreman/api/test_repositories.py [hunk 2]
```diff
@@ -186,6 +191,41 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
...
+        2. Assert that no errors related to compression type  are present in
+            /var/log/messages.
+        3. Assert that sync was executed properly.
+
+    :CaseComponent: Pulp
+
+    :BZ: 2241934
+    """
+    repo = module_target_sat.api.Repository(
+        product=module_product, content_type='yum', url=f'{PULP_FIXTURE_ROOT}/rpm-zstd-metadata'
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repositories.py:217]: Here {PULP_FIXTURE_ROOT}/rpm-zstd-metadata can be added as constant which can be leveraged in future
```suggestion
        product=module_product, content_type='yum', url=FAKE_ZST_REPO
```

File: tests/foreman/api/test_repositories.py [hunk 3]
```diff
@@ -186,6 +191,41 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
...
+    ).create()
+    assert repo.read().content_counts['rpm'] == 0
+    sync = module_product.sync()
+    assert sync['result'] == 'success'
+    assert repo.read().content_counts['rpm'] >= 1
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repositories.py:222]:
```suggestion
    assert repo.read().content_counts['rpm'] > 0
```

File: tests/foreman/api/test_repositories.py [hunk 4]
```diff
@@ -186,6 +191,41 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
...
+    :customerscenario: true
+
+    :steps:
+
+        1. Sync upstream repository having zst type compression.
+        2. Assert that no errors related to compression type  are present in
+            /var/log/messages.
+        3. Assert that sync was executed properly.
+
+    :CaseComponent: Pulp
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repositories.py:212]:
```suggestion
    :CaseComponent: Repositories
```

File: tests/foreman/api/test_repositories.py [hunk 5]
```diff
@@ -186,6 +191,41 @@ def test_positive_sync_kickstart_repo(module_sca_manifest_org, target_sat):
...
+
+    :BZ: 2241934
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repositories.py:214]: It should be added below to the BZ for better readability
```suggestion
    :BZ: 2241934
    
    :customerscenario: true
```

File: tests/foreman/api/test_repositories.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_repositories.py]: Updated

File: tests/foreman/api/test_repositories.py (hunk 3 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_repositories.py]: Updated

File: tests/foreman/api/test_repositories.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_repositories.py]: Updated

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repositories.py -k test_positive_sync_upstream_repo_with_zst_compression

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6162
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_repositories.py -k test_positive_sync_upstream_repo_with_zst_compression --external-logging
Test Result : =========== 1 passed, 10 deselected, 1 warning in 653.39s (0:10:53) ============
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**